wordpress-mobile/WordPress-iOS
# Issue/7081 Media Library: Import media from other applications

This partially addresses [7081](https://github.com/wordpress-mobile/WordPress-iOS/issues/7081)

This feature adds new options to the Media Library > add media functionality:
- Utilizes `UIDocumentPickerViewController` to enable users to add photos and video from other applications. 
- Adding the ability take a photo/video directly.

If iOS11+, tapping the '+' button displays a menu with 3 options: Take a Photo or Video, Photo Library, and Other Apps. (If not iOS11, it falls back to the previous behavior, i.e. immediately launching the Photo Library.)

**Other Apps**

Selecting 'Other Apps' displays the Document Picker with the most recent location. Navigating to Browse > Locations, the user can select different locations (i.e. applications).

![simulator](https://user-images.githubusercontent.com/1816888/31200642-f4b482d8-a918-11e7-9124-ebdb549af713.gif)

**Take Photo or Video**

Selecting 'Take Photo or Video' displays a `WPMediaCapturePresenter` to capture photos or videos. Because this option is now on the menu, `allowCaptureOfMedia` has been disabled in the Photo Library.

**To test:**
- Prerequisite: any device must have external media to select from. Ex: iCloud enabled, Google Drive installed.
- Run the app on iOS11+ devices. Verify the Media Library "+" button presents the menu.
- Verify you can add media from a selected location.
- Verify you can take a photo/video from the camera.
- Run the app on <iOS11 devices. Verify the Media Library "+" button displays the Photo Library.
